### PR TITLE
Typescript vue

### PR DESCRIPTION
**Pull request by Shapus**
Vue files refactotring with using VUEX
This changes do not include css/html fix (except several times where it was necessary)
Email channel verification not working now

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,3 +1,41 @@
 <template>
```
Comment (Shapus) [inline src/App.vue:1]: If we will use .env file, FETCH_CONFIG will be deleted

File: src/App.vue [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+      'login',
+      'logout'
+    ])
+  },
+  created() {
+    //this.$i18n.locale = this.$translate()
+    this.FETCH_CONFIG().then(()=>{
+      this.init = true
+    })
+    if(sessionStorage.getItem('sessionKey')){
```
Comment (Shapus) [inline src/App.vue:29]: FETCH_AVATAR API call using for checking if session key is valid

File: src/components/emailLogin.vue [hunk 3]
```diff
@@ -67,132 +67,95 @@
...
-<script>
-	export default {
+<script lang="ts">
+declare var grecaptcha: any;
+import { defineComponent } from 'vue'
+import { mapGetters, mapActions, mapMutations } from 'vuex';
+	export default defineComponent({
 		data() {
 			return {
-				errors: null,
```
Comment (Shapus) [inline src/components/emailLogin.vue:74]: Don't know why it look like this. Anyway, I think we will use PRETTIER later, so it doesn't matter

File: src/components/locator.vue [hunk 4]
```diff
@@ -17,9 +17,9 @@
                 <div class="mt-1 float-end"><!-- <a class="link-light" href="#">finde mich</a> --></div>
             </div>
             <form class="d-flex" @submit.prevent="submit">
-                <tomSelect ref="ts" v-if="locationsAll" :locations="locationsAll" @tsChanged="showServicesAtLocation"></tomSelect>
```
Comment (Shapus) [inline src/components/locator.vue:20]: Can't do anything about my pedantry)

File: src/components/locator.vue [hunk 5]
```diff
@@ -126,7 +133,8 @@ export default {
                     }
                 );
             };
-            this.mapScript.src = 'https://maps.googleapis.com/maps/api/js?key='+this.map.key+'&language='+this.$i18n.locale;
+            let key = process.env.NODE_ENV=="local"?"":"key="+this.googleMapKey+"&"
```
Comment (Shapus) [inline src/components/locator.vue:136]: Google map api key won't use on local enviroment. Maybe it's bad idea

**Review by zahhar — APPROVED** [review]

**Pull request merged**